### PR TITLE
Ledger: Fix issue with double write-a-head logging trie removal 

### PR DESCRIPTION
**Pull request by ramtinms**
The complete ledger uses HashiCorp LRU cache to keep the tries. 
Any call to LRU cache remove seems to call the evict function, so there is an odd cycle in the ledger when replaying of WALs that will append the delete trie logs to the WAL segments (again)).

### CODE REVIEW

File: ledger/complete/ledger.go [hunk 1]
```diff
@@ -77,11 +77,15 @@ func NewLedger(dbDir string,
...
 	}
 
+	// pause records to prevent double logging trie removals
+	w.PauseRecord()
 	err = w.ReplayOnForest(forest)
 	if err != nil {
 		return nil, fmt.Errorf("cannot restore LedgerWAL: %w", err)
 	}
 
+	w.UnpauseRecord()
```
Comment (Kay-Zee) [inline ledger/complete/ledger.go:87]: could use defer pattern, so that we unpause even if theres err

File: cmd/util/cmd/read-ledger-wals/main.go [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package main
+
+import "github.com/onflow/flow-go/cmd/util/cmd/read-ledger-wals/cmd"
+
+func main() {
+	cmd.Execute()
```
Comment (Kay-Zee) [inline cmd/util/cmd/read-ledger-wals/main.go:6]: does it make sense to have it as it's own cmd? or should we add it to general utils?

File: ledger/complete/wal/wal.go [hunk 3]
```diff
@@ -16,6 +16,7 @@ const SegmentSize = 32 * 1024 * 1024
 
 type LedgerWAL struct {
 	wal            *prometheusWAL.WAL
+	paused         bool
```
Comment (Kay-Zee) [inline ledger/complete/wal/wal.go:19]: not thread safe, but i think that's okay.

**Review by Kay-Zee — APPROVED** [review]
small comment about defer, but other wise LGTM

File: ledger/complete/wal/wal.go (hunk 3 above)
Comment (ramtinms) [reply ledger/complete/wal/wal.go:19]: yeah I think wal is not also threadsafe.

File: ledger/complete/ledger.go (hunk 1 above)
Comment (ramtinms) [reply ledger/complete/ledger.go:89]: done at bb3bcac.

File: cmd/util/cmd/read-ledger-wals/main.go (hunk 2 above)
Comment (ramtinms) [reply cmd/util/cmd/read-ledger-wals/main.go]: I moved it to the read-exec-state as a subcommand.

Comment (ramtinms) [reply cmd/util/cmd/read-ledger-wals/main.go]: applied at bb3bcac

**Review by AlexHentschel — APPROVED** [review]
👍

**Pull request merged**